File: Fourth Semester - Design and Programming Environments/Java/ProiectJavaNetworking/TravelModel/src/java/org/example/model/validator/TourValidator.java
package org.example.model.validator;

import org.example.model.Tour;

public class TourValidator implements Validator<Tour> {
    @Override
    public void validate(Tour entity) {
        if (entity == null) {
            throw new IllegalArgumentException("Tour is null");
        }
        if (entity.getDestination() == null || entity.getDestination().isEmpty()) {
            throw new ValidationException("Destination is empty");
        }
        if (entity.getDepartureDate() == null) {
            throw new ValidationException("Departure date is null");
        }
        if (entity.getNumberOfAvailableSeats() == null) {
            throw new ValidationException("Number of available places is null");
        }
        if (entity.getPrice() < 0) {
            throw new ValidationException("Price is negative");
        }
    }
}

File: Fourth Semester - Design and Programming Environments/C#/ProiectC#Networking/TravelModel/model/validator/TourValidator.cs
namespace TravelModel.model.validator;
public class TourValidator: IValidator<Tour>
{
    public bool Validate(Tour entity)
    {
        if (entity == null)
        {
            throw new ArgumentNullException(nameof(entity), "Tour cannot be null");
        }

        if (string.IsNullOrEmpty(entity.Destination))
        {
            throw new ValidationException("Destination cannot be empty");
        }

        if (string.IsNullOrEmpty(entity.TransportCompany))
        {
            throw new ValidationException("TransportCompany cannot be empty");
        }
        
        if (entity.DepartureDate == null)
        {
            throw new ValidationException("Departure Date cannot be empty");
        }

        if (entity.NumberOfAvailableSeats == null)
        {
            throw new ValidationException("Number Of Available Seats cannot be empty");
        }

        if (entity.Price == null)
        {
            throw new ValidationException("Price cannot be empty");

        }
        return true;
    }
}